lib/cloud/imds/gcp/imds.go

```diff
@@ -121,23 +171,20 @@ func (client *InstanceMetadataClient) GetTags(ctx context.Context) (map[string]s
 	}
 	// Get labels.
 	var gcpLabels map[string]string
-	inst, err := client.instancesClient.GetInstance(ctx, req)
+	inst, err := client.instanceGetter.GetInstance(ctx, req)
 	if err == nil {
 		gcpLabels = inst.Labels
 	} else if trace.IsAccessDenied(err) {
-		client.labelPermissionErrorOnce.Do(func() {
-			slog.WarnContext(ctx, "Access denied to instance labels, does the instance have compute.instances.get permission?")
-		})
```

Comment:
Getting rid of the once here means that we'll get this message every time the node tries to update its labels. Just wanted to make sure that's the effect you intended.

Reply:
You're not missing anything. It sounds like we're good then.